### ISSUE TITLE
LOGLEVEL ERROR by default

### ISSUE THREAD

**Issue by KhaimovMR** [fennerm/flashfocus#32]
By default, when running the flashfocus w/o LOLLEVEL env variable set to 'ERROR' the flashfocus is flooding the stdout with a lot of the INFO level messages. I'd suggest to change the default level from the INFO to the ERROR.
1. This way flashfocus won't spend the precious microtime for the unnecessary logging
2. The stdout won't be flooded for the users, who use the flashfocus as just the utility w/o any intentions to develop it.

PS: Got the PR for this issue: https://github.com/fennerm/flashfocus/pull/33

**Comment by fennerm** [comment]
I realize that logging verbosely by default is not the way things are typically done, but I like it for flashfocus because it lowers the barrier to entry for people to report issues. People should only really be running flashfocus in terminal when they're testing the program out. When its run in an init script the output won't be rendered in terminal and the performance impact should be minimal.

That being said I'm open to this if there's tangible evidence that this is affecting user experience e.g performance monitoring with/without your change.

**Comment by maximbaz** [comment]
Hi @fennerm, I wanted to report the same issue before I found this one. You are suggesting users to run `flashfocus` as `exec_always --no-startup-id flashfocus` from i3/sway, but what this does is it floods i3/sway logs with verbose logging from this app. It might or might not impact the performance, but it certainly makes it more difficult to track real errors, because logs are full of "unimportant" messages.

Maybe set log level to ERROR by default, and add `-v / --verbose` flag that would override it to INFO ?

**Comment by fennerm** [comment]
@maximbaz That's a fair point. How about we try a compromise? I would like for the early init messages to still be logged by default, but I can set the loglevel to DEBUG for the window update messages that flood your logs. For those I can add a `--verbose` option.

Here's an example of the messages which would still be logged by default:
```
INFO: Detected display protocol: wayland - sway
INFO: Loading configuration from /home/fen-arch/.config/flashfocus/flashfocus.yml
INFO: Initializing with parameters:
{'ntimepoints': 10, 'flash_fullscreen': False, 'flash_on_focus': True, 'default_opacity': 1.0, 'flash_opacity': 0.7, 'time': 800.0, 'flash_lone_windows': 'always', 'simple': False, 'rules': None}
INFO: Initializing default window opacity...
INFO: Setting all windows to their default opacity...
INFO: Initializing threads...
INFO: Threads initialized, waiting for events...
^CWARNING: Interrupt received, shutting down...
INFO: Terminating threads...
INFO: Disconnecting socket...
INFO: Resetting windows to full opacity...
INFO: Disconnecting from display server...
```

**Comment by maximbaz** [comment]
It would definitely be better, yes 👍 

Just curious, why do you want to keep these messages? Is it because people are reporting bugs without providing details? If so, maybe a better approach would be to create an issue template which will be asking to paste an output of `flashfocus -v` ?

**Comment by fennerm** [comment]
The hope is that providing more info by default helps people to troubleshoot issues on their own. I think some % of users would just give up on flashfocus before thinking to turn on verbose logging tbh. 

I don't have any data on this so who knows, but if the only downside is a few extra lines in a logfile it seems worth it.

I'll go ahead and just add a `--loglevel` param so you can silence flashfocus completely if you like

**Comment by fennerm** [comment]
@maximbaz --loglevel param is included in the latest release (v2.2.0). Let me know if you have any issues.

**Comment by fennerm** [comment]
@maximbaz Hey heads up I had to change this param to (--verbosity/-v). Dumb error on my  part with the initial naming - https://github.com/fennerm/flashfocus/issues/54
